### PR TITLE
Create text_detection.py

### PR DESCRIPTION
**Pull request by vishwesh5**
#12270 #13429
**Deep Learning text detection sample (Python)**
- Tested on **Ubuntu 18.04** - OpenCV 3.4.3, OpenCV 3.4.4, OpenCV 4.0 (master branch)
- Python version supported - Python 2 and Python 3

### CODE REVIEW

**Comment by vishwesh5** [comment]
@dkurt I have made changes to the code as you suggested (except the arguments). 
@berak Thanks for your help. I have fixed the trailing whitespace issue.

File: samples/dnn/text_detection.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            offset = ([offsetX + cosA * x1_data[x] + sinA * x2_data[x], offsetY - sinA * x1_data[x] + cosA * x2_data[x]])
+            # Find points for rectangle
+            p1 = (-sinA * h + offset[0], -cosA * h + offset[1])
+            p3 = (-cosA * w + offset[0],  sinA * w + offset[1])
+            center = (0.5*(p1[0]+p3[0]), 0.5*(p1[1]+p3[1]))
+            detections.append((center, (w,h), -1*angle * 180.0 / math.pi))
+            confidences.append(float(score))
+    # Return detections and confidences
+    return [detections, confidences]
+def main():
```
[the remark this answers is not in the document]
Comment (vishwesh5) [reply samples/dnn/text_detection.py]: And done! I really hope I didn't make some new mistake this time.

**Comment by dkurt** [comment]
@vishwesh5, please note I changes the base branch from `master` to `3.4`. You need to update it locally:

```
git checkout master
git br -D patch-1
git fetch origin patch-1
git checkout patch-1
```

**Comment by vishwesh5** [comment]
@dkurt done! Thanks for your help!

File: samples/dnn/text_detection.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import numpy as np
+
+############ Add argument parser for command line arguments ############
+parser = argparse.ArgumentParser(description='Use this script to run text detection deep learning networks using OpenCV.')
+# Input argument
+parser.add_argument('--input', help='Path to input image or video file. Skip this argument to capture frames from a camera.')
+# Model argument
+parser.add_argument('--model', required=True,
+                    help='Path to a binary .pb file of model contains trained weights.'
+                    )
```
Comment (dkurt) [inline samples/dnn/text_detection.py]: What about these comments? I still think that `parser.add_argument('--model'` is enough and there is no need to replicate it in comments.

Please also do not put `)` to the new lines

```python
help='Path to a binary .pb file of model contains trained weights.')
```
looks fine.

File: samples/dnn/text_detection.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
+# Import required modules
+import cv2 as cv
+import math
+import argparse
+import time
```
Comment (dkurt) [inline samples/dnn/text_detection.py]: Unused package.

Comment (vishwesh5) [reply samples/dnn/text_detection.py]: Done!

File: samples/dnn/text_detection.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
+# Import required modules
+import cv2 as cv
+import math
+import argparse
+
+############ Add argument parser for command line arguments ############
+parser = argparse.ArgumentParser(description='Use this script to run text detection deep learning networks using OpenCV.')
```
Comment (dkurt) [inline samples/dnn/text_detection.py]: It'd be nice to add some references to origin paper and network's repo. I.e. like in text_detection.cpp:

```
Use this script to run TensorFlow implementation (https://github.com/argman/EAST) of
EAST: An Efficient and Accurate Scene Text Detector (https://arxiv.org/abs/1704.03155v2)
```

**Comment by vishwesh5** [comment]
Sure. I will make that addition in a couple of hours (I am traveling right
now)

**Comment by vishwesh5** [comment]
Thank you for your help!

**Pull request merged**